src/Migrate.js

```diff
@@ -0,0 +1,169 @@
+import { InputState, TokenAmountInput, TokenSelect } from "./Inputs";
+import {
+  BASE_FACTOR,
+  ELC,
+  LIQUIDITY_FEE,
+  TOKENS,
+  USD,
+  ZERO,
+} from "./constants";
+import {
+  encodeToken,
+  formatTokenBalance,
+  tokenTicker,
+  findToken,
+  Value,
+  formatCurrency,
+} from "./helpers";
+import { usePostTransaction } from "./mutations";
+import { find, get } from "lodash";
+import { ExchangeCalculator } from "ellipticoin";
+import { default as React, useMemo, useState, useEffect, useRef } from "react";
+import { Button, Form, Collapse } from "react-bootstrap";
+import { ArrowDown } from "react-feather";
+import { ChevronLeft } from "react-feather";
+import nacl from "tweetnacl";
+import base64url from "base64url";
+import { getMessage } from "eip-712";
+import { ethers } from "ethers";
+import cbor from "cbor";
+import EthCrypto from "eth-crypto";
+import { actions } from "ellipticoin";
+
+import {
+  useEthereumAccounts,
+  useEthereumConnected,
+  getAccounts,
+} from "./ethereum";
+import { useLocalStorage } from "./helpers";
+
+export default function Migrate() {
+  const [secretKey, setSecretKey] = useLocalStorage(
+    "secretKey",
+    () => nacl.sign.keyPair().secretKey
+  );
+  const publicKey = useMemo(
+    () => Buffer.from(nacl.sign.keyPair.fromSecretKey(secretKey).publicKey),
+    [secretKey]
+  );
+  const accounts = useEthereumAccounts();
+
+  const step = () => {
+    if (accounts[0]) {
+      return (
+        <SignAndMigrate
+          address={accounts[0]}
+          secretKey={secretKey}
+          publicKey={publicKey}
+        ></SignAndMigrate>
+      );
+    } else if (window.ethereum) {
+      return (
+        <Button
+          onClick={() => ethereum.request({ method: "eth_requestAccounts" })}
+        >
+          Connect to MetaMask
+        </Button>
+      );
+    } else {
+      return <InstallMetamask />;
+    }
+  };
+  return (
+    <div className="section">
+      <div className="appHeader no-border transparent position-absolute">
+        <div className="pageTitle"></div>
+        <div className="right"></div>
+      </div>
+      <div id="appCapsule" className="">
+        <div className="section text-center">
+          <h1>Migration Required</h1>
+        </div>
+        <div className="row justify-content-md-center">
+          <div className="card col col-8">
+            <div className="card-body">
+              <p style={{ color: "red", textAlign: "center" }}>
+                Ellipticoin is transitioning from native addresses and private
+                keys to peer chain addresses and private keys. The first peer
+                chain integration is Ethereum. Your private key is from the
+                Ellipticoin prototype. Please proceed with migration by clicking
+                below.
```

Comment:
It seems to me that this summarizes the "big change" that this branch, and the corresponding one in the backend repo, implement. If I have this correct, the change is that now I could have an Ethereum wallet somewhere else, and now I could upload that key pair into ellipticoin and it would register my value from my Eth wallet? Or perhaps it simply now acknowledges the existence of that Eth wallet and allows me to transition value from it directly into the Ellipticoin network?

Reply:
Yep! That's the big change. Originally we had our own private keys and address formatting. We wanted to make it super easy to get started so we generated a private key for you when you first loaded the wallet.

As we're observing users use the wallet we're realizing people are much more interested in using this as a trading tool than they are in using it as a Venmo replacement.

Since that's the case we're transitioning to ETH addresses and private keys.

This gives us a few things:

1. We no longer have to be responsible for users private keys. Multiple people have already lost their keys and switching to ETH private keys allows us to offload that task to MetaMask. Whenever we want a signature we just request one from MetaMask and boom the transaction is signed.

2. This makes the barrier to entry for exiting ETH users much lower. If they have Metamask all they have to do is make one transaction to get onto the Ellipticoin network.